### ISSUE TITLE
Errors being thrown during chunk generation

### ISSUE THREAD

**Issue by Anaerin** [ricksouth/serilum-mc-mods#1064]
## **Information**
Minecraft version: 1.18.2
Modloader: Forge
Forge version: 40.1.16  OR  Fabric loader version: x.xx.x
Environment: Multiplayer

Mod name: No Hostiles Around Campfire
Mod version: 3.9

## **Description**
When generating chunks, an error is being thrown sometimes in the log (naturally generating a campfire in a chunk?)

## **Crash report**
```
[10May2022 17:20:35.289] [Server thread/ERROR] [net.minecraft.world.level.levelgen.structure.PoolElementStructurePiece/]: Input d  oes not contain a key [element_type]: MapLike[{}]
[10May2022 17:20:35.290] [Server thread/ERROR] [net.minecraft.world.level.levelgen.structure.pieces.PiecesContainer/]: Exception   loading structure piece with id minecraft:jigsaw
java.lang.IllegalStateException: Invalid pool element found
        at net.minecraft.world.level.levelgen.structure.PoolElementStructurePiece.m_163129_(PoolElementStructurePiece.java:54) ~[  server-1.18.2-20220404.173914-srg.jar%23285!/:?]
        at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
        at net.minecraft.world.level.levelgen.structure.PoolElementStructurePiece.<init>(PoolElementStructurePiece.java:54) ~[ser  ver-1.18.2-20220404.173914-srg.jar%23285!/:?]
        at net.minecraft.world.level.levelgen.structure.pieces.PiecesContainer.m_192753_(PiecesContainer.java:73) ~[server-1.18.2  -20220404.173914-srg.jar%23285!/:?]
        at net.minecraft.world.level.levelgen.feature.StructureFeature.m_191128_(StructureFeature.java:124) ~[server-1.18.2-20220  404.173914-srg.jar%23285!/:?]
        at net.minecraft.world.level.chunk.storage.ChunkSerializer.m_188254_(ChunkSerializer.java:464) ~[server-1.18.2-20220404.1  73914-srg.jar%23285!/:?]
        at net.minecraft.world.level.chunk.storage.ChunkSerializer.m_188230_(ChunkSerializer.java:196) ~[server-1.18.2-20220404.1  73914-srg.jar%23285!/:?]
        at net.minecraft.server.level.ChunkMap.m_203107_(ChunkMap.java:543) ~[server-1.18.2-20220404.173914-srg.jar%23285!/:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:157) ~[server-1.18.2-20220404.173914-srg.  jar%23285!/:?]
        at net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_6367_(ServerChunkCache.java:532) ~[server-1.18.2-2022  0404.173914-srg.jar%23285!/:?]
        at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:131) ~[server-1.18.2-20220404.173914-srg.  jar%23285!/:?]
        at net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_7245_(ServerChunkCache.java:540) ~[server-1.18.2-2022  0404.173914-srg.jar%23285!/:?]
        at net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:140) ~[server-1.18.2-20220404.173914-srg  .jar%23285!/:?]
        at net.minecraft.server.level.ServerChunkCache.m_7587_(ServerChunkCache.java:129) ~[server-1.18.2-20220404.173914-srg.jar  %23285!/:?]
        at net.minecraft.world.level.Level.m_6522_(Level.java:169) ~[server-1.18.2-20220404.173914-srg.jar%23285!/:?]
        at net.minecraft.world.level.LevelReader.m_46819_(LevelReader.java:123) ~[server-1.18.2-20220404.173914-srg.jar%23285!/:?  ]
        at net.minecraft.world.level.Level.m_6325_(Level.java:164) ~[server-1.18.2-20220404.173914-srg.jar%23285!/:?]
        at com.natamus.collective.functions.FABFunctions.getBlockEntitiesAroundPosition(FABFunctions.java:33) ~[collective-1.18.2  -4.25.jar%2381!/:4.25]
        at com.natamus.collective.functions.FABFunctions.getAllTileEntityPositionsNearbyEntity(FABFunctions.java:48) ~[collective  -1.18.2-4.25.jar%2381!/:4.25]
        at com.natamus.nohostilesaroundcampfire.events.CampfireEvent.onEntityJoin(CampfireEvent.java:107) ~[nohostilesaroundcampf  ire_1.18.2-3.9.jar%23223!/:1.0]
        at net.minecraftforge.eventbus.ASMEventHandler_1646_CampfireEvent_onEntityJoin_CheckSpawn.invoke(.dynamic) ~[?:?]
        at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-5.0.3.jar%232!/:?]
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!/:?]
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!/:?]
        at net.minecraftforge.event.ForgeEventFactory.canEntitySpawn(ForgeEventFactory.java:191) ~[forge-1.18.2-40.1.16-universal  .jar%23289!/:?]
        at net.minecraftforge.common.ForgeHooks.canEntitySpawn(ForgeHooks.java:1093) ~[forge-1.18.2-40.1.16-universal.jar%23289!/  :?]
        at net.minecraft.world.level.NaturalSpawner.m_47038_(NaturalSpawner.java:183) ~[server-1.18.2-20220404.173914-srg.jar%232  85!/:?]
        at net.minecraft.world.level.NaturalSpawner.m_47045_(NaturalSpawner.java:126) ~[server-1.18.2-20220404.173914-srg.jar%232  85!/:?]
        at net.minecraft.world.level.NaturalSpawner.m_47029_(NaturalSpawner.java:116) ~[server-1.18.2-20220404.173914-srg.jar%232  85!/:?]
        at net.minecraft.server.level.ServerChunkCache.m_8490_(ServerChunkCache.java:355) ~[server-1.18.2-20220404.173914-srg.jar  %23285!/:?]
        at net.minecraft.server.level.ServerChunkCache.m_201698_(ServerChunkCache.java:309) ~[server-1.18.2-20220404.173914-srg.j  ar%23285!/:?]
        at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:281) ~[server-1.18.2-20220404.173914-srg.jar%23285!/:?  ]
        at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:902) ~[server-1.18.2-20220404.173914-srg.jar%23285!/  :?]
        at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:326) ~[server-1.18.2-20220404.173914-srg.j  ar%23285!/:?]
        at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:842) ~[server-1.18.2-20220404.173914-srg.jar%23285!/  :?]
        at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:685) ~[server-1.18.2-20220404.173914-srg.jar%23285  !/:?]
        at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261) ~[server-1.18.2-20220404.173914-srg.jar%23285  !/:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]
```

**Comment by Anaerin** [comment]
This is still happening with Forge 40.1.68, and is causing major tick lag as well: https://spark.lucko.me/OyZHSYQr9f

**Comment by ricksouth** [comment]
Thank you for giving the mod a try and opening the issue. This has been fixed in version 7.0 of No Hostiles Around Campfire. Apologies it took so long for me to get to.

https://curseforge.com/minecraft/mc-mods/no-hostiles-around-campfire/files
